### ISSUE TITLE
[Question] about using transaction objects

### ISSUE THREAD

**Issue by atari83** [jtv/libpqxx#701]
Hello!

1- I'm using `pqxx::read_transaction` for the queries that only read/fetche data and doesnt make any changes to db. Did I chose correct transaction class for such usage ? and then I noticed there is `commit()` function in the class, do i still need to call even it doesnt change anything ?

2- Also I'm using `pqxx::work` for the queries (either i directly use insert/update or just calling a function), prepared-statements or etc that modifies the records in db. I noticed that `auto_commit` is already ON, so do I still need to call `commit()` function ?

Thanks.

**Comment by jtv** [comment]
Yes, `read_transaction` is the right class.  There is no big difference, but you may feel safer knowing that these transactions will make no changes to the database.

And as you noticed, when you make no changes to the database there's no real need to commit the transaction.

When you're in a transaction though, you are not in autocommit.  In libpqxx, the only way to work in autocommit is by using `nontransaction`, which has the API of a transaction class but doesn't actually start a transaction on the database.
